### ISSUE TITLE
[openssl] build failure by emscripten

### ISSUE THREAD

**Issue by hushanjushi** [microsoft/vcpkg#29575]
**Host Environment**

- OS: Windows 11
- Compiler: Emcc 3.1.31

**To Reproduce**

`./vcpkg install openssl --triplet wasm32-emscripten`

**Failure logs**

```
-- Building wasm32-emscripten-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: C:/Environment/CMake/bin/cmake.exe --build . --config Debug --target install -- -v -j9
    Working Directory: C:/Environment/vcpkg/buildtrees/openssl/wasm32-emscripten-dbg
    See logs for more information:
      C:\Environment\vcpkg\buildtrees\openssl\install-wasm32-emscripten-dbg-out.log
```

```shell
[1/3] cmd.exe /C "cd /D C:\Environment\vcpkg\buildtrees\openssl\wasm32-emscripten-dbg\nssl-3.0.7-96d825e305.clean && set PATH=C:/Environment/emsdk/upstream/emscripten;%PATH% && set CC=C:/Environment/emsdk/upstream/emscripten/emcc.bat && set AR=C:/Environment/emsdk/upstream/emscripten/emar.bat && set LD=C:/Environment/msys64/ucrt64/bin/ld.lld.exe && set RANLIB=C:/Environment/emsdk/upstream/emscripten/emranlib.bat && set MAKE=C:/Environment/emsdk/upstream/emscripten/emmake && set MAKEDEPPROG=C:/Environment/emsdk/upstream/emscripten/emcc.bat && set WINDRES= && C:\Environment\emsdk\upstream\emscripten\emconfigure ./config no-shared no-module threads no-engine no-dso no-asm no-shared no-sse2 no-srtp --prefix=C:/Environment/vcpkg/packages/openssl_wasm32-emscripten/debug --openssldir=/etc/ssl --cross-compile-prefix=\"/\" && C:\Environment\CMake\bin\cmake.exe -DDIR=C:/Environment/vcpkg/buildtrees/openssl/wasm32-emscripten-dbg/nssl-3.0.7-96d825e305.clean -P C:/Environment/vcpkg/ports/openssl/unix/remove-deps.cmake"
FAILED: nssl-3.0.7-96d825e305.clean/Makefile C:/Environment/vcpkg/buildtrees/openssl/wasm32-emscripten-dbg/nssl-3.0.7-96d825e305.clean/Makefile 
cmd.exe /C "cd /D C:\Environment\vcpkg\buildtrees\openssl\wasm32-emscripten-dbg\nssl-3.0.7-96d825e305.clean && set PATH=C:/Environment/emsdk/upstream/emscripten;%PATH% && set CC=C:/Environment/emsdk/upstream/emscripten/emcc.bat && set AR=C:/Environment/emsdk/upstream/emscripten/emar.bat && set LD=C:/Environment/msys64/ucrt64/bin/ld.lld.exe && set RANLIB=C:/Environment/emsdk/upstream/emscripten/emranlib.bat && set MAKE=C:/Environment/emsdk/upstream/emscripten/emmake && set MAKEDEPPROG=C:/Environment/emsdk/upstream/emscripten/emcc.bat && set WINDRES= && C:\Environment\emsdk\upstream\emscripten\emconfigure ./config no-shared no-module threads no-engine no-dso no-asm no-shared no-sse2 no-srtp --prefix=C:/Environment/vcpkg/packages/openssl_wasm32-emscripten/debug --openssldir=/etc/ssl --cross-compile-prefix=\"/\" && C:\Environment\CMake\bin\cmake.exe -DDIR=C:/Environment/vcpkg/buildtrees/openssl/wasm32-emscripten-dbg/nssl-3.0.7-96d825e305.clean -P C:/Environment/vcpkg/ports/openssl/unix/remove-deps.cmake"
configure: ./config no-shared no-module threads no-engine no-dso no-asm no-shared no-sse2 no-srtp --prefix=C:/Environment/vcpkg/packages/openssl_wasm32-emscripten/debug --openssldir=/etc/ssl --cross-compile-prefix="/"
emconfigure: error: './config no-shared no-module threads no-engine no-dso no-asm no-shared no-sse2 no-srtp --prefix=C:/Environment/vcpkg/packages/openssl_wasm32-emscripten/debug --openssldir=/etc/ssl --cross-compile-prefix="/"' failed: [WinError 193] %1 ������Ч�� Win32 Ӧ�ó���
ninja: build stopped: subcommand failed.
```

**Additional context**

Add any other context about the problem here, such as what you have already tried to resolve the issue.

**Comment by dg0yt** [comment]
Can you test #27261?
(I didn't test with emscripten, but I tried to remove some oddities.)

**Comment by hushanjushi** [comment]
> Can you test #27261? (I didn't test with emscripten, but I tried to remove some oddities.)

I changed to x64-linux as host OS, emscripten seems to depend on OS in some levels.

**Comment by dg0yt** [comment]
> > Can you test #27261? (I didn't test with emscripten, but I tried to remove some oddities.)
> 
> I changed to x64-linux as host OS, emscripten seems to depend on OS in some levels.

Would you be so kind to test it anyways? There is no CI for emscripten, and you would be affected if that PR breaks this platform.

**Comment by hushanjushi** [comment]
> > > Can you test #27261? (I didn't test with emscripten, but I tried to remove some oddities.)
> > 
> > 
> > I changed to x64-linux as host OS, emscripten seems to depend on OS in some levels.
> 
> Would you be so kind to test it anyways? There is no CI for emscripten, and you would be affected if that PR breaks this platform.

I tested. fail, too.

**Comment by dg0yt** [comment]
Should be fixed by #29966. But please test yourself.

**Comment by hushanjushi** [comment]
> Should be fixed by #29966. But please test yourself.

Work, thank you very much.
